<launch>
  <arg name="open_rviz" default="false"/>  

  <!-- offb_node -->
  <node pkg="mav_slam_nav" type="mav_slam_nav_vel" name="mav_slam_nav_vel_node" output="screen">
  </node>

  <!-- slam -->
  <node pkg="gmapping" name="slam_gmapping" type="slam_gmapping">   
     <remap from="/scan" to="/scan"/> 
     <param name="map_frame" value="map" />
     <param name="odom_frame" value="odom" />
  </node>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find mav_slam_nav)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find mav_slam_nav)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find mav_slam_nav)/param/local_costmap_params_test.yaml" command="load" />
    <rosparam file="$(find mav_slam_nav)/param/global_costmap_params_test.yaml" command="load" />
    <rosparam file="$(find mav_slam_nav)/param/move_base_params.yaml" command="load" />  
    <rosparam file="$(find mav_slam_nav)/param/dwa_local_planner_params.yaml" command="load" />
    <remap from="cmd_vel" to="/cmd_vel"/>
    <remap from="odom" to="/mavros/odometry/in"/>
  </node>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find mav_slam_nav)/rviz/mav_slam_nav.rviz"/>
  </group>
</launch>